workflows/Workflow.Samples/Execute.cs

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https:*www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START workflows_api_quickstart]
+
+using Google.Cloud.Workflows.Common.V1;
+using Google.Cloud.Workflows.Executions.V1;
+using System;
+using System.Threading.Tasks;
+
+public class ExecuteWorkflowSample
+{
+    /// <summary>
+    /// Execute a workflow and return the execution operation.
+    /// </summary>
+    /// <param name="projectID">Your Google Cloud Project ID.</param>
+    /// <param name="locationID">The region where your workflow is located.</param>
+    /// <param name="workflowID">Your Google Cloud Workflow ID.</param>
+    /// 
+    public async Task<Execution> ExecuteWorkflow(
+        string projectId = "YOUR-PROJECT-ID",
+        string locationID = "YOUR-LOCATION-ID",
+        string workflowID = "YOUR-WORKFLOW-ID"
+    )
+    {
+        // Initialize the client.
+        ExecutionsClient client = await ExecutionsClient.CreateAsync();
+
+        // Build the parent location path.
+        string parent = WorkflowName.Format(projectId, locationID, workflowID);
+
+        // Craete te execution request.
+        CreateExecutionRequest createExecutionRequest = new CreateExecutionRequest
+        {
+            ParentAsWorkflowName = WorkflowName.Parse(parent),
+        };
+
+        // Execute the operation.
+        Execution execution = await client.CreateExecutionAsync(createExecutionRequest);
+        Console.WriteLine("- Execution started...");
+
+        Execution fetchedExecution;
+        DateTime startTime = DateTime.UtcNow;
+
+        // TODO(developer): Adjust the following time parameters according to your Workflow timeout settings.
+        int fetchDelayMilliseconds = 1000;
+        TimeSpan timeout = TimeSpan.FromMinutes(5);
+
+        // Loop to check whether the execution is done or the timeout has been reached.
+        do
+        {
+            fetchedExecution = await client.GetExecutionAsync(execution.Name);
+
+            if (DateTime.UtcNow - startTime > timeout)
+            {
+                Console.WriteLine($"Timeout reached after {timeout}");
```

Comment:
You'll print this even  if the latest fetchedExecution has completed, which seems unexpected.

You can use a cancellation token that you obtaing from a timespan based [cancellation token source](https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.-ctor).

Reply:
And do we have to show the loop? Changes are that these executions are meant to be offline operations that take longer (than 5 minutes) to execute.

Can we show instead how to check the current state o the excution by calling GetExecution once?